aws/resource_aws_redshift_scheduled_action.go

```diff
@@ -0,0 +1,316 @@
+	conn := meta.(*AWSClient).redshiftconn
+	name := d.Get("name").(string)
+	createOpts := &redshift.CreateScheduledActionInput{
+		ScheduledActionName: aws.String(name),
+		Schedule:            aws.String(d.Get("schedule").(string)),
+		IamRole:             aws.String(d.Get("iam_role").(string)),
+		TargetAction:        expandRedshiftScheduledActionTargetAction(d.Get("target_action")),
+	}
+	if attr, ok := d.GetOk("description"); ok {
+		createOpts.ScheduledActionDescription = aws.String(attr.(string))
+	}
+	if attr, ok := d.GetOk("active"); ok {
+		createOpts.Enable = aws.Bool(attr.(bool))
+	}
+	if attr, ok := d.GetOk("start_time"); ok {
+		t, _ := time.Parse(time.RFC3339, attr.(string))
+		createOpts.StartTime = aws.Time(t)
+	}
+	if attr, ok := d.GetOk("end_time"); ok {
+		t, _ := time.Parse(time.RFC3339, attr.(string))
+		createOpts.EndTime = aws.Time(t)
+	}
+
+	log.Printf("[DEBUG] Creating Redshift Scheduled Action: %s", createOpts)
+
+	// Retry for IAM eventual consistency
+	err := resource.Retry(1*time.Minute, func() *resource.RetryError {
+		_, err := conn.CreateScheduledAction(createOpts)
+
+		// InvalidParameterValue: If you create iam role same time, you must wait the role will be valid
+		if isAWSErr(err, "InvalidParameterValue", "The IAM role must delegate access to Amazon Redshift scheduler") {
+			return resource.RetryableError(err)
+		}
+
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		return fmt.Errorf("error creating Redshift Scheduled Action (%s): %s", name, err)
+	}
+
+	d.SetId(name)
+
+	return resourceAwsRedshiftScheduledActionRead(d, meta)
+}
+
+func resourceAwsRedshiftScheduledActionRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).redshiftconn
+	name := d.Id()
+
+	descOpts := &redshift.DescribeScheduledActionsInput{
+		ScheduledActionName: aws.String(name),
+	}
+
+	resp, err := conn.DescribeScheduledActions(descOpts)
+	if err != nil {
+		return fmt.Errorf("Error describing Redshift Scheduled Action %s: %s", d.Id(), err)
+	}
+
+	if resp.ScheduledActions == nil || len(resp.ScheduledActions) != 1 {
+		log.Printf("[WARN] Unable to find Redshift Scheduled Action (%s)", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	scheduledAction := resp.ScheduledActions[0]
+
+	d.Set("name", scheduledAction.ScheduledActionName)
+	d.Set("description", scheduledAction.ScheduledActionDescription)
+	d.Set("active", scheduledAction.State)
+	d.Set("schedule", scheduledAction.Schedule)
+	d.Set("iam_role", scheduledAction.IamRole)
+
+	if aws.TimeValue(scheduledAction.StartTime).Format(time.RFC3339) != "0001-01-01T00:00:00Z" {
+		d.Set("start_time", aws.TimeValue(scheduledAction.StartTime).Format(time.RFC3339))
+	}
+
+	if aws.TimeValue(scheduledAction.EndTime).Format(time.RFC3339) != "0001-01-01T00:00:00Z" {
+		d.Set("end_time", aws.TimeValue(scheduledAction.EndTime).Format(time.RFC3339))
+	}
```

Comment:
If each of StartTime and EndTime are not set, "0001-01-01T00:00:00Z" was returned. So when it was returned, start_time and end_time are not set.

Reply:
You may be able to check against `scheduledAction.StartTime != nil` and `scheduledAction.EndTime != nil` instead 👍

Reply:
@bflad Thank you for telling, I fixed it!